src/lib/pmd/CommandLineSupport.ts

```diff
@@ -0,0 +1,66 @@
+import { Logger } from '@salesforce/core';
+import {AsyncCreatable} from '@salesforce/kit';
+import {ChildProcessWithoutNullStreams} from 'child_process';
+import cspawn = require('cross-spawn');
+
+export abstract class CommandLineSupport extends AsyncCreatable {
+
+	private parentLogger: Logger;
+	private parentInitialized: boolean;
+
+	protected async init(): Promise<void> {
+
+		if (this.parentInitialized) {
+			return;
+		}
+
+		this.parentLogger = await Logger.child('CommandLineSupport');
+		this.parentInitialized = true;
+	}
+
+	protected abstract buildClasspath(): Promise<string[]>;
+
+	/**
+	 * Accepts a child process created by child_process.spawn(), and a Promise's resolve and reject functions.
+	 * Resolves/rejects the Promise once the child process finishes.
+	 * @param cp
+	 * @param res
+	 * @param rej
+	 */
+	protected monitorChildProcess(cp: ChildProcessWithoutNullStreams, res: (string) => void, rej: (string) => void): void {
```

Comment:
It looks like this method is still defined in some of the child classes. Should those be annotated as overrides?

Reply:
Good catch. I don't think the overrides do anything different. We'll need to remove them. @gjung-sfdc fyi

Reply:
@rmohan20 @jfeingold35 monitorChildProcess is ever so slightly different in PmdCatalogWrapper so I didn't want to mess anything up by using the version in CommandLineSupport. Should I have them all use the same version or annotate as an override?